tlslite/utils/python_tripledes.py

```diff
@@ -0,0 +1,546 @@
+#################################################
+#               Documentation                   #
+#################################################
+
+# Author:   Todd Whiteman
+# Date:     16th March, 2009
+# Verion:   2.0.0
+# License:  Public Domain - free to do as you wish
+# Homepage: http://twhiteman.netfirms.com/Des.html
+#
+# Modified by: Adam Varga, 2018
+#
+# A pure python implementation of the DES and Triple DES
+# encryption algorithms using CBC mode. Triple DES class is
+# implemented by utilising the DES base. Triple DES is
+# DES-EDE3 with a 24 byte key, or DES-EDE2 with a 16 byte key.
+
+"""
+Class initialization
+--------------------
+pyDes.Des(key, [iv])
+pyDes.Triple_des(key, [iv])
+
+key -> Bytes containing the encryption key. 8 bytes for DES, 16 or 24 bytes
+       for Triple DES
+iv  -> Optional Initial Value bytes, must be supplied if using CBC mode.
+       Length must be 8 bytes.
+"""
+
+import sys
+
+# PY_VER is used to handle Python2 and Python3 differences.
+PY_VER = (sys.version_info[0], )
+
+def new(key, iv=None):
+    # the new() method to operate this cipher
+    return Python_TripleDES(key, iv)
+
+class _baseDes(object):
+    # The base class shared by des and triple des.
+    def __init__(self, iv=None):
+        if iv:
+            iv = self._guard_against_unicode(iv)
+        self.block_size = 8
+
+        # Sanity checking of arguments.
+        if iv and len(iv) != self.block_size:
+            raise ValueError("Invalid Initial Value (iv), must be a "
+                             "multiple of {0} bytes".format(self.block_size))
+        self._iv = iv
+
+    @property
+    def key(self):
+        # return bytes
+        return self.__key
+
+    @key.setter
+    def key(self, key):
+        # Set the crypting key for this object
+        key = self._guard_against_unicode(key)
+        self.__key = key
+
+    @property
+    def iv(self):
+        # return bytes
+        return self._iv
+
+    @iv.setter
+    def iv(self, iv):
+        # Set the Initial Value
+        if not iv or len(iv) != self.block_size:
+            raise ValueError("Invalid Initial Value (iv), must be a "
+                             "multiple of {0} bytes".format(self.block_size))
+        iv = self._guard_against_unicode(iv)
+        self._iv = iv
+
+    def _guard_against_unicode(self, data):
+        # Only accept byte strings or ascii unicode values, otherwise
+        # there is no way to correctly decode the data into bytes.
+        if PY_VER < (3, ):
```

Comment:
either `if PY_VER < (3, )` or `PY_VER = (sys.version_info[0], )` ([line 33](https://github.com/tomato42/tlslite-ng/pull/202/files#diff-592bf4889628685f27f43910cc5f6a0aR33)) both at the same time won't work